### DISCUSSION TITLE
useImportExtensions should support forcing .js extension for sourcecode imports

### DISCUSSION THREAD

**Discussion by kibertoad** [biomejs/biome#2966] [Ideas]
Currently `useImportExtensions` rule proposed fix uses a real file extension, which in case of TypeScript is ".ts". This is not correct.
See this discussion:
https://stackoverflow.com/questions/62619058/appending-js-extension-on-relative-import-statements-during-typescript-compilat

Specifically this comment from a person from TS team:
"TS does not re-write import paths. This something the TS team has been adamant about it. Write the paths that work at runtime, and if needed configure ts to make those paths work. ( in this case no configuration is needed)"

**Comment by ematipico** [comment]
This seems to be more like a bug. Can you report it?

**Reply by kibertoad** [reply]
Done!
https://github.com/biomejs/biome/issues/2967
